yesod-test/test/main.hs

```diff
@@ -145,21 +145,25 @@ main = hspec $ do
                 request $ do
                     setMethod "POST"
                     setUrl $ LiteAppRoute ["post"]
-                    addPostParam "foo" "foobarbaz"
+                    -- If value uses special characters,
+                    addPostParam "foo" "foo+bar%41<&baz"
                 statusIs 200
-                bodyEquals "foobarbaz"
+                -- They pass through the server correctly.
+                bodyEquals "foo+bar%41<&baz"
             yit "labels" $ do
                 get ("/form" :: Text)
                 statusIs 200
 
                 request $ do
                     setMethod "POST"
                     setUrl ("/form" :: Text)
-                    byLabel "Some Label" "12345"
+                    byLabel "Some Label" "foo+bar%41<&baz"
                     fileByLabel "Some File" "test/main.hs" "text/plain"
```

Comment:
Here's the test case that passes through special characters with multipart/form-data. (This test did not fail previously, so the bug I'm fixing here is only with form-url-encoded. I thought it failed at one point, because the endpoint this test is accessing returns HTML instead fo text, so `&` comes through as `&amp;` etc.

Reply:
Got it, thanks for clarifying.